tests/main/interfaces-cups/task.yaml

```diff
@@ -0,0 +1,66 @@
+summary: Ensure that the cups interfaces work with app providers
+
+details: |
+    A snap providing the cups-control and cups interfaces should be able to
+    create the control socket, with connecting consuming snaps able to use it.
+    This intentionally does not test the mediation properties of the cupsd
+    server.
+
+prepare: |
+    snap pack test-snapd-provider
+    snap install --dangerous ./test-snapd-provider_1_all.snap
+    snap pack test-snapd-consumer
+    snap install --dangerous ./test-snapd-consumer_1_all.snap
+
+    if [ -e /run/cups ]; then
+        mv /run/cups /run/cups.orig
```

Comment:
fun trick, in master you can now do this in prepare:

```bash
tests.cleanup prepare
tests.cleanup defer mv /run/cups.orig /run/cups
```

and then in restore do

```bash
tests.cleanup restore
```

but I won't block on this

Reply:
We also have fs-state helpers that handle the backup/restore case directly. Both can be used together defer restore